MachObfuscator/DependencyAnalysis/ObfuscableFilesFilter.swift

```diff
@@ -11,6 +11,21 @@ extension ObfuscableFilesFilter {
         }
     }
 
+    func negate() -> ObfuscableFilesFilter {
+        return ObfuscableFilesFilter { !self.isObfuscable($0) }
+    }
+
+    func whitelist(_ other: ObfuscableFilesFilter) -> ObfuscableFilesFilter {
+        return ObfuscableFilesFilter { url in
+            // Order of checks is important.
```

Comment:
I don't understand why is this order important. From the mathematical point of view this doesn't matter, does it?
`whitelist` will always win with the blacklist.

Reply:
Whitelist must be evaluated before blacklist and it should short-circuit further checks. ObfuscableFilesFilter expression tree building is currently very simple, so this can be obtained by one of the following:
1) requiring that command line arguments are given in specific order: first whitelist then blacklist
2)grouping all whitelisted elements first and blacklisted at the end of the expression.

I chose the second option, so you can pass `--skip-all-frameworks` and `--obfuscate-framework` in any order.

Reply:
After more thinking and testing I realised, that previous implementation was incorrect. I fixed it. Now order is not so important (only last check `onlyFiles` must be `and`'ed with combined results of all other checks so we do not modify system files), but grouping is.

In general, order of boolean expressions is important:
`(blacklist1 || whitelist1) && blacklist2` is not equivalent to `(blacklist1 && blacklist2) || whitelist1)` (parentheses added for clarity)

Reply:
Now it looks much better :) Thanks!